### PR TITLE
Add return and where clause

### LINKED ISSUE #9: Support for WHERE keyword
Add support for `WHERE` keyword
https://neo4j.com/docs/cypher-manual/current/clauses/where/
```
MATCH (n)
WHERE n.name = 'Peter' XOR (n.age < 30 AND n.name = 'Timothy') OR NOT (n.name = 'Timothy' OR n.name = 'Peter')
RETURN n.name, n.age
```

### PR DESCRIPTION
**Pull request by martinemdal**
Added a missing return clause rather than allowing for it to be implicit in any query context. Also added a naive where clause with a greater than function. Closes #9

### CODE REVIEW

File: src/main/kotlin/net/emdal/tank/clause/WhereClause.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import net.emdal.tank.IntProperty
+import net.emdal.tank.Property
+import net.emdal.tank.StringProperty
+
+class WhereClause(
+  private val alias: String,
+  override var query: List<String> = listOf("WHERE "),
+  override var aliases: Map<String, List<Entity>> = emptyMap()
+) : Clause {
+  infix fun Property.greaterThan(value: Any) = WhereClause(
```
Comment (martinemdal) [inline src/main/kotlin/net/emdal/tank/clause/WhereClause.kt]: Should greater than be allowed for all `Property`s. Might not make sense for string or other types of properties. This should be a function of `IntProperty`.

File: src/main/kotlin/net/emdal/tank/clause/ReturnClause.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package net.emdal.tank.clause
+
+import net.emdal.tank.Entity
+
+class ReturnClause(
+  override var query: List<String> = listOf("RETURN "),
+  override var aliases: Map<String, List<Entity>> = emptyMap()
```
Comment (martinemdal) [inline src/main/kotlin/net/emdal/tank/clause/ReturnClause.kt]: Aliases are not in use anywhere. The type inference between the string alias and the entity types do no carry on between clauses which means this is most likely useless to keep adding to the clauses. I think we should remove the `aliases` field from Clause and rather add it back in if needed for something in the future.

**Comment by martinemdal** [comment]
Also removed the aliases map from Clauses as it is not currently being used for anything.

File: src/main/kotlin/net/emdal/tank/clause/WhereClause.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package net.emdal.tank.clause
+
+import net.emdal.tank.IntProperty
+
+class WhereClause(
+  private val alias: String,
+  override var query: List<String> = listOf("WHERE ")
```
Comment (kollstrom) [inline src/main/kotlin/net/emdal/tank/clause/WhereClause.kt]: Should this be a part of the constructor

**Review by kollstrom — CHANGES_REQUESTED** [review]

**Review by kollstrom — APPROVED** [review]

**Comment by kollstrom** [comment]
@martinemdal Should we try to get this merged soon?

**Pull request merged**